ompi/mpi/c/reduce_local.c

```diff
@@ -76,8 +79,12 @@ int MPI_Reduce_local(const void *inbuf, void *inoutbuf, int count,
     /* Invoke the op component to perform the back-end operation */
     OBJ_RETAIN(op);
     /* XXX -- CONST -- do not cast away const -- update mca/coll */
-    ompi_op_reduce(op,(void *) inbuf, inoutbuf, count, datatype);
+    // Since there is no 'comm' parameter to this interface use 'self' since
+    // this is a local operation to this process.
+    ompi_communicator_t *comm = &ompi_mpi_comm_self.comm;
+    err = comm->c_coll.coll_reduce_local(inbuf, inoutbuf, count, datatype, op,
+                                         comm->c_coll.coll_reduce_local_module);
     OBJ_RELEASE(op);
 
-    return MPI_SUCCESS;
+    OMPI_ERRHANDLER_RETURN(err, MPI_COMM_WORLD, err, FUNC_NAME);
```

Comment:
I think the problem is that users who set error handlers rarely set them on MPI_COMM_SELF, so I suppose what you have is "fine" (rather than changing the param checking lines to MPI_COMM_Self which would be more correct).

The OBJ_RETAIN / OBJ_RELEASE is a bug waiting to happen, but it does take time, and I doubt anyone but a test case would actually do that.  And when it does we can handle that all in a different PR.